include/simple-mpc/friction-compensation.hpp

```diff
@@ -0,0 +1,56 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 2-Clause License
+//
+// Copyright (C) 2025, INRIA
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+#pragma once
+
+#include <pinocchio/fwd.hpp>
+// Include pinocchio first
+#include <Eigen/Dense>
+
+#include "simple-mpc/fwd.hpp"
+#include <pinocchio/multibody/model.hpp>
+
+namespace simple_mpc
+{
+  using namespace pinocchio;
+
+  /**
+   * @brief Class managing the friction compensation for torque
+   *
+   * It applies a compensation term to a torque depending on dry and
+   * viscuous frictions.
+   */
+  class FrictionCompensation
+  {
+  public:
+    /**
+     * @brief Construct a new Friction Compensation object
+     *
+     * @param actuation_size Dimension of torque
+     */
+    FrictionCompensation(const Model & model, const long actuation_size);
+
+    /**
+     * @brief Compute the torque correction due to friction and store it internally.
+     *
+     * @param[in] velocity Joint velocity
+     * @param[in] torque Joint torque
+     */
+    void computeFriction(const Eigen::VectorXd & velocity, const Eigen::VectorXd & torque);
```

Comment:
The torque could be a non-const ref (maybe eigen réf should used) , so that it used as a return value, which could prevent some copies etc..

Also, since it is not used for the friction estimation, the torque vector could be completely overwritten by this method and contain only the delta needed to be applied to the real torque.

Reply:
(You'll have to create a small proxy function for the python bindings, to create an eigen vector and pass it to this method)

Reply:
I'm on it